src/routes/admin.routes.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express')
+const router = new express.Router()
+// const { app } = require('./routes')
+
+const User = require('../db/userModel')
+const auth = require('../middlewares/auth')
+const isAdmin = require('../middlewares/admin')
+
+router.get('/users', auth, isAdmin, async (req, res) => {
+    try {
+        const users = await User.find({})
+        if (!users) return res.sendStatus(400)
+        res.render('adminView', {
+            users
+        })
+    } catch (error) {
+        res.sendStatus(500)
+    }
+})
+
+router.post('/users/update', auth, isAdmin, async (req, res) => {
+    try {
+        const { id, role } = req.body
```

Comment:
Good use of try-catch for error handling. However, the catch block could be more informative by logging the error or sending a more descriptive response.

Reply:
The check if (!users) might not behave as expected if User.find({}) returns an empty array, which is truthy. Consider explicitly checking if users.length is zero.

Reply:
Model.findOne, If no document satisfies the query, the method returns null is what I understand, hence the check might not be needed.